src/App.tsx

```diff
@@ -1,26 +1,133 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import { FC, Dispatch, SetStateAction, useEffect, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import { getFilteredTodos } from './utils/getFilteredTodos';
+
+import { createTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotifications';
+import { TodoItemTemp } from './components/TodoItemTemp';
+
+import { Todo } from './types/Todo';
+import { Errors } from './types/Errors';
+import { FilterStatus } from './types/FilterStatus';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoToDelete, setTodoToDelete] = useState<number[]>([]);
```

Comment:
```suggestion
  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
```
It stores IDs, not todos as I see, and it's probably better to have a more general name as you can also change status of todo as well, not only delete